orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileManagerImpl.java

```diff
@@ -382,50 +382,63 @@ public OrcidProfile createOrcidProfileAndNotify(OrcidProfile orcidProfile) {
+
+                if (existingProfileEntity != null) {
+                    // Dont delete the existing elements anymore
+                    // profileDao.removeChildrenWithGeneratedIds(existingProfileEntity);
+                    setWorkPrivacy(orcidProfile, defaultVisibility);
+                    setAffiliationPrivacy(orcidProfile, defaultVisibility);
+                    setFundingPrivacy(orcidProfile, defaultVisibility);
+                }
+                dedupeWorks(orcidProfile);
+                dedupeAffiliations(orcidProfile);
+                dedupeFundings(orcidProfile);
+                addSourceToEmails(orcidProfile, existingProfileEntity, amenderOrcid);
 
-        ProfileEntity profileEntity = adapter.toProfileEntity(orcidProfile, existingProfileEntity, updateOptions);
-        profileEntity.setLastModified(new Date());
-        profileEntity.setIndexingStatus(IndexingStatus.PENDING);
-        profileDao.flush();
-        ProfileEntity updatedProfileEntity = profileDao.merge(profileEntity);
-        profileDao.refresh(updatedProfileEntity);
-        OrcidProfile updatedOrcidProfile = convertToOrcidProfile(updatedProfileEntity, LoadOptions.ALL);
+                Boolean claimed = orcidProfile.getOrcidHistory() != null ? orcidProfile.getOrcidHistory().isClaimed() : existingProfileEntity.getClaimed();
 
-        orcidProfileCacheManager.put(updatedOrcidProfile);
-        return updatedOrcidProfile;
+                if (orcidProfile.getOrcidInternal() != null && orcidProfile.getOrcidInternal().getPreferences() != null
+                        && orcidProfile.getOrcidInternal().getPreferences().getActivitiesVisibilityDefault() != null) {
+                    defaultVisibility = orcidProfile.getOrcidInternal().getPreferences().getActivitiesVisibilityDefault().getValue();
+                }
+                addDefaultVisibilityToBioItems(orcidProfile, defaultVisibility, claimed);
+
+                ProfileEntity profileEntity = adapter.toProfileEntity(orcidProfile, existingProfileEntity, updateOptions);
+                profileEntity.setLastModified(new Date());
+                profileEntity.setIndexingStatus(IndexingStatus.PENDING);
+                profileDao.flush();
+                ProfileEntity updatedProfileEntity = profileDao.merge(profileEntity);
+                profileDao.refresh(updatedProfileEntity);
+                
+                // Then persist the works
+                if(orcidProfile.getOrcidActivities() != null && orcidProfile.getOrcidActivities().getOrcidWorks() != null) {
+                    adapter.setWorks(profileEntity, orcidProfile.getOrcidActivities().getOrcidWorks());
+                } else {
+                    adapter.setWorks(profileEntity, null);
+                }
+            }
+        });
+        
+        OrcidProfile profile = transactionTemplate.execute(new TransactionCallback<OrcidProfile>() {
```

Comment:
And then we open another transaction to fetch the latest OrcidProfile with the latest changes